### DISCUSSION TITLE
help (adding text under lyrics)

### DISCUSSION THREAD

**Discussion by WEJ19** [opensheetmusicdisplay/opensheetmusicdisplay#965] [Q&A] [answered]
I want to add text boxes under every voice lyric, could you point me in the right direction?  Do you think it would be easier to approach this problem where you generate the graphical lyric position or after the music sheet has been returned?

Alternatively, could I make page breaks at the end of each system to make space to add in text boxes in a webpage?

Thank you for your hard work.

**Comment by sschmidTU** [comment, marked as answer]
Have you tried just adding text to the lyrics?

```js
// before osmd.render(), after load()
for (const measures of osmd.graphic.measureList) {
  for (const measure of measures) {
    for (const staffEntry of measure.staffEntries) {
      for (const lyricsEntry of staffEntry.LyricsEntries) {
        lyricsEntry.text += "\n\ntest"; // line breaks may have to be done differently than "\n".
      }
    }
  }
}
```

alternatively, you can get the boundingbox (PositionAndShape) of the GraphicalLyricsEntry, then draw text below it
(`osmd.drawer.backend.renderText()`, or create a GraphicalLabel and use `osmd.drawer.renderLabel()`)
